### PR TITLE
`General`: Don't show "No description available" in course header

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The header of exercises and courses should be small and don't take up unnecessary screen space. Therefore unavailable and not completely necessary information should be left out.

### Description

If a course has no description, it's not shown in the course header any more instead of "No description available".

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Go to a course with no description
3. Check that there is no unnecessary string
4. Add a description to the course
5. Check that it's correctly displayed

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Old:
Bildschirmfoto 2022-06-10 um 16 18 25
New:
Bildschirmfoto 2022-06-10 um 16 21 56

### CODE REVIEW

**Review by ge65cer — CHANGES_REQUESTED** [review]
Please remove the extra margin on the bottom when no description is shown.

Screenshot 2022-06-10 at 17 06 11

**Review by nadinesamyy — COMMENTED** [review]
"translation-not-found[descriptionTest]" was displayed instead of the text, not sure if this is an error from the ts or if it is related to the PR but I tested on ts1/ts3

**Comment by ge65cer** [comment]
Thanks for fixing the margin. If the course description is an empty string, the description should also be hidden, shouldn't it?

**Comment by ge65cer** [comment]
Thanks for fixing the margin. If the course description is an empty string (whitespace), the description should also be hidden, shouldn't it?

**Comment by JohannesStoehr** [comment]
>  If the course description is an empty string (whitespace), the description should also be hidden, shouldn't it?

Currently it is not possible set the description to an empty string. And multiple whitespaces should rather get filtered out when saving the changed course than in the exercise header. So I would handle this in this PR

**Review by willergeorg — DISMISSED** [review]
Tested on TS5. I agree with @ge65cer, that invalid descriptions (e.g. only spaces), should also be omitted. If this is out of scope of this PR, please provide a follow-up as soon as possible. Code lgtm as well. :)

File: src/main/webapp/app/course/manage/detail/course-detail.component.html [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
                 <div class="col general-info">
-                    <h3 class="fw-medium">{{ course.title }}</h3>
-                    <h4 class="fw-medium">{{ courseDescription || ('artemisApp.courseOverview.noDescription' | artemisTranslate) }}</h4>
-                    <h6 class="fw-medium" role="button" *ngIf="enableShowMore && !longDescriptionShown" (click)="toggleCourseDescription()">
-                        {{ 'artemisApp.courseOverview.showMore' | artemisTranslate }}
-                    </h6>
-                    <h6 class="fw-medium" role="button" *ngIf="enableShowMore && longDescriptionShown" (click)="toggleCourseDescription()">
-                        {{ 'artemisApp.courseOverview.showLess' | artemisTranslate }}
-                    </h6>
+                    <h3 class="fw-medium" [ngStyle]="courseDescription ? {} : { 'margin-bottom': 0 }">{{ course.title }}</h3>
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/course/manage/detail/course-detail.component.html:6]:
```suggestion
                    <h3 class="fw-medium" [class.mb-0]="!courseDescription">{{ course.title }}</h3>
```
Should also work (please test this first :D) ->  https://angular.io/guide/class-binding

File: src/main/webapp/app/overview/course-overview.component.html [hunk 2]
```diff
@@ -3,14 +3,16 @@
...
                 <div class="col general-info">
-                    <h3 class="fw-medium">{{ course.title }}</h3>
-                    <h4 class="fw-medium">{{ courseDescription || ('artemisApp.courseOverview.noDescription' | artemisTranslate) }}</h4>
-                    <h6 class="fw-medium" role="button" *ngIf="enableShowMore && !longDescriptionShown" (click)="toggleCourseDescription()">
-                        {{ 'artemisApp.courseOverview.showMore' | artemisTranslate }}
-                    </h6>
-                    <h6 class="fw-medium" role="button" *ngIf="enableShowMore && longDescriptionShown" (click)="toggleCourseDescription()">
-                        {{ 'artemisApp.courseOverview.showLess' | artemisTranslate }}
-                    </h6>
+                    <h3 class="fw-medium" [ngStyle]="courseDescription ? {} : { 'margin-bottom': 0 }">{{ course.title }}</h3>
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/overview/course-overview.component.html:6]: same as above :)

**Review by Dominik-Weinzierl — COMMENTED** [review]
Some minor suggestions

**Review by jschmautzer — APPROVED** [review]
tested on ts1, works as described

**Review by TheTrueVester — DISMISSED** [review]
Tested on ts1. Everything worked as expected.

**Review by alaraozdenler — DISMISSED** [review]
Tested on ts2, works as described

**Review by babyygemperor — DISMISSED** [review]
Tested on ts1, works as described

**Review by jbllhff — DISMISSED** [review]
Tested on ts2, works great!

**Review by Dominik-Weinzierl — APPROVED** [review]
code 👍

**Review by mallgaier — APPROVED** [review]
Tested, works as expected. Good job 👍

**Review by ge65cer — APPROVED** [review]
Approved, please keep the follow-up PR in mind.

**Pull request merged**